tls/test_tls_stress.py

```diff
@@ -1,41 +1,98 @@
 """
-TLS Stress tests.
+TLS Stress tests - load Tempesta FW with multiple TLS connections.
 """
-
 import os
+
 from helpers import control, tf_cfg
-from testers import stress
+from framework import tester
 
 __author__ = 'Tempesta Technologies, Inc.'
-__copyright__ = 'Copyright (C) 2018 Tempesta Technologies, Inc.'
+__copyright__ = 'Copyright (C) 2018-2019 Tempesta Technologies, Inc.'
 __license__ = 'GPL2'
 
 
-class StressTls(stress.StressTest):
-    """Load Tempesta with multiple TLS connections.
-    """
+class StressTls(tester.TempestaTest):
+
+    backends = [
+        {
+            'id' : '0',
+            'type' : 'nginx',
+            'check_ports' : [
+                {
+                    "ip" : "${server_ip}",
+                    "port" : "8000",
+                }
+            ],
+            'status_uri' : 'http://${server_ip}:8000/nginx_status',
+            'config' : """
+                pid ${pid};
+                worker_processes    auto;
+                events {
+                    worker_connections 1024;
+                    use epoll;
+                }
+                http {
+                    keepalive_timeout ${server_keepalive_timeout};
+                    keepalive_requests ${server_keepalive_requests};
+                    sendfile        on;
+                    tcp_nopush      on;
+                    tcp_nodelay     on;
+                    open_file_cache max=1000;
+                    open_file_cache_valid 30s;
+                    open_file_cache_min_uses 2;
+                    open_file_cache_errors off;
+                    error_log /dev/null emerg;
+                    access_log off;
+                    server {
+                        listen       ${server_ip}:8000;
+                        location / {
+                            return 200;
+                        }
+                        location /nginx_status {
+                            stub_status on;
+                        }
+                    }
+                }
+            """,
+            }
+        ]
 
-    defconfig = ("tls_certificate %s/tfw-root.crt;\n"
-                 "tls_certificate_key %s/tfw-root.key;\n"
-                 "listen 443 proto=https;\n\n")
+    clients = [
+        {
+            'id' : '0',
+            'type' : 'wrk',
+            'addr' : "${tempesta_ip}:443",
+            'ssl' : True,
+        },
+    ]
 
-    def create_clients(self):
-        wrk = control.Wrk(ssl=True)
+    def setUp(self):
+        dir_path = os.path.dirname(os.path.abspath(__file__))
+        self.tempesta = {
+            'config' : """
+                listen 443 proto=https;
+                tls_certificate %s/tfw-root.crt;
+                tls_certificate_key %s/tfw-root.key;
+                server ${server_ip}:8000;
+            """ % (dir_path, dir_path),
+        }
+        tester.TempestaTest.setUp(self)
```

Comment:
I see, why it was chosen to dynamically fill Tempesta's configuration file: idea was to use paths to files with certificate. But it's a wrong idea. Tempesta and the framework itself are started on the same VM only in developers environment and never on CI. So it's not possible to address the files directly in the configuration. It's necessary to copy the certificate from the framework vm to Tempesta's vm before test start.

See `self.tempesta.node.copy_file_to_node(cert % dir, cert % path)` calls in the original tests.

Reply:
Fixed.